### ISSUE TITLE
Drop kFrag consistency check (i.e., vkeys stuff)? 

### ISSUE THREAD

**Issue by cygnusv** [nucypher/pyUmbral#77]
Before that I came up with the zero-knowledge verification protocol, the kFrag consistency thing was the best thing we had. But now it seems not very useful. Let's recall what it achieves.

vKeys are produced by Alice during split re-encryption key generation, and distributed to all Ursulas together with each kFrag, so they can check whether their kFrag is consistent with the rest. "Consistent" here means that all kFrags encode a common shared secret. However, this doesn't check anything regarding the actual validity of this shared secret, which in our case is a re-encryption key.

So it's possible right now that a cheating Alice creates a consistent set of kFrags so the consistency check passes on each Ursula. However, this will be detected by our fancy zk-validation protocol once Bob sees that decryption fails on his side. 

It seems then that the only way it is useful is to detect situations where there is a small subset of erroneous kFrags (which may be intentionally or not), but that the rest are OK.

**Comment by tuxxy** [comment]
This will remove `challenge` and `check_challenge`, correct?

**Comment by cygnusv** [comment]
This would remove:
- `KFrag.is_consistent()`
- Parameter `h` in `UmbralParameters`
- The code in `split_rekey()` associated to `vkeys` generation

**Comment by tuxxy** [comment]
Are you gonna take this, or should I? I can have a PR in tomorrow (mine)
morning.

**Comment by cygnusv** [comment]
I can do it, but of course I need that @jMyles and @michwill agree on this. BTW, what's your opinion?

**Comment by tuxxy** [comment]
I completely agree with it. I think the rest of us do, too. I raised this
issue on Discord a while back since we can't/don't use the vKeys.

**Comment by cygnusv** [comment]
OK, I'm working on some stuff in my fork, so once I have confirmation by the rest, then I'll make the changes and make a PR.

**Comment by jMyles** [comment]
@cygnusv: indeed, @tuxxy is correct (at least in my case) when he says we're already on board for this change - my apologies for not weighing-in with approval earlier.  This will be great.  :-)
